src/Data/Managers/V2/GetMetadata/useGetMetadata.ts

```diff
@@ -0,0 +1,72 @@
+import { useQuery } from "@tanstack/react-query";
+import { ParsedNumber } from "@/Core";
+import { PrimaryBaseUrlManager } from "@/UI";
+import { useFetchHelpers } from "../helpers";
+
+/**
+ * Custom hook to fetch metadata for a specific service entity and key.
+ * @param environment - The environment of the service.
+ * @param service_entity - The entity name of the service.
+ * @param service_id - The ID of the service.
+ * @param instanceVersion - The version of the service instance.
+ * @param key - The key for the metadata.
+ * @returns An object containing the custom hook.
+ */
+export const useGetMetadata = (
+  environment: string,
+  service_entity: string,
+  service_id: string,
+  instanceVersion: ParsedNumber | undefined,
+  key: string,
+) => {
+  const baseUrlManager = new PrimaryBaseUrlManager(
+    globalThis.location.origin,
+    globalThis.location.pathname,
+  );
+  const { createHeaders, handleErrors } = useFetchHelpers();
+  const headers = createHeaders(environment);
+  const baseUrl = baseUrlManager.getBaseUrl(process.env.API_BASEURL);
+
+  const getMetadata = async (): Promise<{
+    data: string | undefined;
```

Comment:
maybe would be good to have a dedicated type for this? Like we have in most cases?

Reply:
added